File: examples/python/ofac_screening.py
"""
InstaTools API - OFAC Sanctions Screening Example

This example demonstrates how to screen entities against the OFAC sanctions list.
"""

import os
import requests

API_KEY = os.getenv('INSTATOOLS_API_KEY', 'YOUR_API_KEY')
BASE_URL = 'https://api.instatools.co'


def screen_entity(name: str, country: str, entity_type: str = 'individual') -> dict:
    """Screen a single entity against OFAC sanctions list."""
    response = requests.post(
        f'{BASE_URL}/api/v1/screen',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={
            'name': name,
            'country': country,
            'entityType': entity_type
        }
    )
    response.raise_for_status()
    return response.json()


def screen_bulk(entities: list) -> dict:
    """Screen multiple entities in bulk (Enterprise tier only)."""
    response = requests.post(
        f'{BASE_URL}/api/v1/screen/bulk',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={'entities': entities}
    )
    response.raise_for_status()
    return response.json()


def main():
    try:
        # Single entity screening
        print('Screening single entity...')
        result = screen_entity('John Doe', 'US')
        print(f'Result: {result}')

        # Check for matches
        if result.get('match'):
            print('WARNING: Potential sanctions match found!')
            print(f"Match score: {result.get('score')}")
        else:
            print('No sanctions matches found.')

        # Bulk screening example (Enterprise tier)
        print('\nBulk screening...')
        bulk_results = screen_bulk([
            {'name': 'Jane Smith', 'country': 'UK'},
            {'name': 'Bob Johnson', 'country': 'CA'}
        ])
        print(f'Bulk Results: {bulk_results}')

    except requests.HTTPError as e:
        print(f'HTTP Error: {e.response.status_code} - {e.response.text}')
    except Exception as e:
        print(f'Error: {e}')


if __name__ == '__main__':
    main()

File: examples/javascript/ofac-screening.js
/**
 * InstaTools API - OFAC Sanctions Screening Example
 * 
 * This example demonstrates how to screen entities against the OFAC sanctions list.
 */

const API_KEY = process.env.INSTATOOLS_API_KEY || 'YOUR_API_KEY';
const BASE_URL = 'https://api.instatools.co';

/**
 * Screen a single entity against OFAC sanctions list
 */
async function screenEntity(name, country, entityType = 'individual') {
  const response = await fetch(`${BASE_URL}/api/v1/screen`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      name,
      country,
      entityType
    })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API Error: ${error.title} - ${error.detail}`);
  }

  return response.json();
}

/**
 * Screen multiple entities in bulk (Enterprise tier only)
 */
async function screenBulk(entities) {
  const response = await fetch(`${BASE_URL}/api/v1/screen/bulk`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ entities })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API Error: ${error.title} - ${error.detail}`);
  }

  return response.json();
}

// Example usage
async function main() {
  try {
    // Single entity screening
    console.log('Screening single entity...');
    const result = await screenEntity('John Doe', 'US');
    console.log('Result:', JSON.stringify(result, null, 2));

    // Check for matches
    if (result.match) {
      console.log('WARNING: Potential sanctions match found!');
      console.log('Match score:', result.score);
    } else {
      console.log('No sanctions matches found.');
    }

    // Bulk screening example (Enterprise tier)
    console.log('\nBulk screening...');
    const bulkResults = await screenBulk([
      { name: 'Jane Smith', country: 'UK' },
      { name: 'Bob Johnson', country: 'CA' }
    ]);
    console.log('Bulk Results:', JSON.stringify(bulkResults, null, 2));

  } catch (error) {
    console.error('Error:', error.message);
  }
}

main();
